evm-contracts/src/v0.6/dev/FluxAggregator.sol

```diff
@@ -99,6 +98,7 @@ contract FluxAggregator is AggregatorV3Interface, Owned {
   uint32 internal latestRoundId;
   mapping(address => OracleStatus) private oracles;
   mapping(uint32 => Round) internal rounds;
+  mapping(uint32 => RoundDetails) internal details;
```

Comment:
This is interesing. Why is this cheaper? I really don't know

Reply:
Solidity will handle packing structs into as few storage slots as possible and when reading efficiently unpacking all of the attributes without doing an expensive storage read for each attribute on the struct. When it does this it is relatively naive and pulls back all attributes of the struct whether they are used or not.

In `getRoundData`, which is the hot path where consumers read data, we were pulling back all of the round details. The cases where we need round details are a subset so we broke them out and can now opt-in to accessing them when needed.